### PR TITLE
Get all branches of a GitRepository

### PR DESCRIPTION
**Pull request by MHudak**
Added functionality to get all branches from a GitHub repository. 

Refactored some of the code:
    renamed getBranch as getDefaultBranch
    fixed typo: paresAsArray -> parseAsArray
    moved formatAPIUrl into new method

This can be used for further functionality such as allowing the user to dynamically choose which branch to show commits for instead of forcing them to choose a static branch when creating the SCM widget.

### CODE REVIEW

**Comment by tabladrum** [comment]
Are we not getting the commits from the other branches?

**Comment by MHudak** [comment]
Not in this PR. We don't need all the commits for each branch in our use case and wanted to get an idea of PR turnaround time before committing additional effort.

File: UI/bower_BACKUP_4748.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{
```
Comment (satishc1) [inline UI/bower_BACKUP_4748.json]: @MHudak Looks like there are some bower jsons that need to be taken out from the PR.

**Comment by satishc1** [comment]
@MHudak Looks like there are a few compile errors due to merging other PRs. Please rebase.

**Comment by nireeshT** [comment]
@satishc1 @tabladrum  LGTM

**Comment by satishc1** [comment]
@MHudak @tabladrum Closing this PR because it doesn't align with the GitHubRepo model which currently works on a specific branch.

The recommendation is create a separate collection called gitmetadata.  This will be updated from GitHubCollectorTask. This is similar to gitrequests which contains pulls and issues and is updated from GitHubCollectorTask.

You can store repo level information inside the gitmetadata collection. Things you could choose to store in here includes branches, last time a branch was updated, releases, commits, contributors etc. The screenshots below show what could be captured.

image

image

**Pull request closed without merging**